pkg/skaffold/util/port_test.go

```diff
@@ -75,3 +75,13 @@ func TestGetAvailablePort(t *testing.T) {
 		t.Errorf("available port (%d) couldn't be used: %w", port, err)
 	}
 }
+
+func TestSystemPorts(t *testing.T) {
+	ps := PortSet{}
+	ps.lock.Lock() // cause deadlock: getPortIfAvailable should never check the PortSet
```

Comment:
not sure why this is required. There's already an error if `getPortIfAvailable` returns true. Also if I deliberately introduce a bug then this deadlock causes the test to hang till timeout.

Reply:
This is a whitebox test that verifies that Skaffold should never check the set-ports-list for system ports.  The intent is that if this test hangs (e.g., someone violated this invariant), someone will quickly discover that this invariant has been violated.

Let me update this comment to be more clear.

Reply:
but won't that get tested anyway in `t.Errorf(...)`? Also can't ps just be `nil` and you get the same thing?

Reply:
I really do want to explicitly verify that `GetPortIfAvailable()` never considers ports 1–1023 separately from the provided `PortSet`.  The `PortSet` uses locks to protect the data structure, and there's no reason to check it for this port range.